src/main/java/datawave/util/time/DateHelper.java

```diff
@@ -121,45 +155,153 @@ public static String formatToTimeExactToSeconds(long inMillis) {
+    /**
+     * Converts a string in a custom format to a Date object in a consistent way not dependent on local settings for calendar, timezone, or locale by using Zulu
+     * timezone and US locale.
+     *
+     * @param date
+     * @param pattern
+     * @return the {@code Date} object
+     */
+    public static Date parseCustom(String date, String pattern) {
+        return lenientParseHelper(date, DateTimeFormatter.ofPattern(pattern).withZone(ZoneOffset.UTC), pattern, true);
+    }
+    
     /**
      * Converts a String in yyyyMMddHH format to a Date object in a consistent way not dependent on local settings for calendar, timezone, or locale by using
      * Zulu timezone and US locale.
-     * 
+     *
      * @param date
      * @return the {@code Date} object
      */
     public static Date parseHour(String date) {
         return lenientParseHelper(date, DTF_hour, DATE_FORMAT_STRING_TO_HOUR, true);
     }
     
+    /**
+     * Converts a string in EEE MM d HH:mm:ss zzz yyyy format to a Date object in a consistent way not dependent on local settings for calendar, timezone, or
+     * locale by using Zulu timezone and US locale.
+     *
+     * @param date
+     * @return the {@code Date} object
+     */
+    public static Date parseDetails(String date) {
+        return lenientParseHelper(date, DTF_Details, DATE_FORMAT_STRING_TO_DETAILS, true);
+    }
+    
+    /**
+     * Return a string representing the given date in yyyy-MM-dd 'T' HH:mm:ss.SSSZ format in a consistent way not dependent on local settings for calendar,
+     * timezone, or locale by using Zulu timezone and US locale
+     *
+     * @param date
+     * @return
+     */
+    public static Date parseFull(String date) {
+        return lenientParseHelper(date, DTF_Full, DATE_FORMAT_STRING_TO_FULL, true);
+    }
+    
+    public static Date parsePurge(String date) {
+        return lenientParseHelper(date, DTF_Purge, DATE_FORMAT_PURGE_CONSTANT, true);
+    }
+    
     /**
      * Only use this for formats that can allow for leniency (i.e. not ISO standard formats).
      */
     private static Date lenientParseHelper(String date, DateTimeFormatter parser, String formatStr, boolean hasTime) {
-        String lenientDate = convertToLenient(date, formatStr);
-        if (hasTime) {
-            return Date.from(ZonedDateTime.parse(lenientDate, parser).toInstant());
-        } else {
-            return Date.from(LocalDate.parse(lenientDate, parser).atStartOfDay(parser.getZone()).toInstant());
+        try {
+            String lenientDate = convertToLenient(date, formatStr);
+            if (hasTime) {
+                return Date.from(ZonedDateTime.parse(lenientDate, parser).toInstant());
+            } else {
+                return Date.from(LocalDate.parse(lenientDate, parser).atStartOfDay(parser.getZone()).toInstant()); // Fails testFailIfMissingDigits test
+            }
+        } catch (DatawaveDateTimeParseException e) {
+            e.printStackTrace();
```

Comment:
printing a stack trace to stderr is not appropriate.  Firstly we should be logging this.  Secondly, why aren't we letting the exception propogate up?